src/main/java/org/broadinstitute/consent/http/models/User.java

```diff
@@ -92,49 +92,45 @@ public User(GoogleUser googleUser) {
 
     /**
      * Convenience method for backwards compatibility support for older clients.
+     * This method is intended to reconstruct a User object from a supplied JSON string
+     * for primary fields and roles. Other associated objects are not intended to be parsed
+     * using this method since it comes from user-supplied data and may conflict with what
+     * the system knows about those associations.
      *
      * @param json A json string that may or may not be correctly structured as a DACUser
      */
     public User(String json) {
         Gson gson = new Gson();
         JsonObject userJsonObject = gson.fromJson(json, JsonObject.class);
-        // Create Date can come in differently, either as a long or a string.
-        // Handle known cases here.
-        String createDateFieldName = "createDate";
-        Date createDate = null;
-        if (userJsonObject.has(createDateFieldName)) {
-            JsonElement createDateElement = userJsonObject.get(createDateFieldName);
-            try {
-                createDate = new Date(createDateElement.getAsLong());
-            } catch (NumberFormatException nfe) {
-                // Known date formats createDate could be in:
-                //   * "Oct 28, 2020"
-                SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy");
-                try {
-                    createDate = sdf.parse(createDateElement.getAsString());
-                } catch (Exception e) {
-                    LoggerFactory
-                        .getLogger(this.getClass())
-                        .error("Unable to parse create date: " + e.getMessage());
-                }
-            }
-            // Remove this from the JSON so we don't re-process it in `gson.fromJson(String, User)`
-            userJsonObject.remove(createDateFieldName);
-        }
-        User u = gson.fromJson(userJsonObject.toString(), User.class);
+        // There are no cases where we want to pull the create date/update date from user-provided data.
+        // Nor do we need to retrieve the full institution object from user-provided data.
+        JsonObject filteredUserJsonObject = filterFields(userJsonObject, Arrays.asList("createDate", "institution"));
```

Comment:
Can you add `libraryCards` to the list just to be safe?

Reply:
Yes, will add that, thanks